src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -576,7 +578,20 @@ private static void EnsureNullableAttributeExists(CSharpCompilation compilation,
                         break;
 
                     case SyntaxKind.ReadOnlyKeyword:
-                        diagnostics.Add(ErrorCode.ERR_ReadOnlyNotSuppAsParamModDidYouMeanIn, modifier.GetLocation());
+                        if (seenReadonly)
+                        {
+                            addERR_DupParamMod(diagnostics, modifier);
+                        }
+                        else if (!seenRef || previousModifier?.Kind() != SyntaxKind.RefKeyword)
```

Comment:
>else if (!seenRef || previousModifier?.Kind() != SyntaxKind.RefKeyword) [](http://example.com/codeflow?start=24&length=71)

This is related to the comment for ```RefReadonlyWithParams_01``` unit-test. We probably shouldn't check ```!seenRef``` here. Also, we probably should change the following ```else``` to ```else if (seenRef)``` and remove the assert in its the block.